subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultArtifactTransformDependencies.java

```diff
@@ -16,68 +16,30 @@
- */
-public class DefaultArtifactTransformDependencies implements ArtifactTransformDependencies {
-    private final ComponentArtifactIdentifier artifactId;
-    private final ResolvableDependencies resolvableDependencies;
-    private final ImmutableAttributes attributes;
+public class DefaultArtifactTransformDependencies implements ArtifactTransformDependenciesInternal {
+    private final FileCollection files;
+    private CurrentFileCollectionFingerprint fingerprint;
 
-    public DefaultArtifactTransformDependencies(ComponentArtifactIdentifier artifactId, ResolvableDependencies resolvableDependencies, ImmutableAttributes attributes) {
-        this.artifactId = artifactId;
-        this.resolvableDependencies = resolvableDependencies;
-        this.attributes = attributes;
+    public DefaultArtifactTransformDependencies(FileCollection files) {
+        this.files = files;
     }
 
     @Override
     public Iterable<File> getFiles() {
-        ResolutionResult resolutionResult = resolvableDependencies.getResolutionResult();
-        Set<ComponentIdentifier> dependenciesIdentifiers = Sets.newHashSet();
-        for (ResolvedComponentResult component : resolutionResult.getAllComponents()) {
-            if (component.getId().equals(artifactId.getComponentIdentifier())) {
-                getDependenciesIdentifiers(dependenciesIdentifiers, component.getDependencies());
-            }
-        }
-        return resolvableDependencies.artifactView(conf -> {
-            conf.componentFilter(element -> {
-                return dependenciesIdentifiers.contains(element);
-            });
-            if (!attributes.isEmpty()) {
-                conf.attributes(container -> {
-                    for (Attribute attribute : attributes.keySet()) {
-                        container.attribute(attribute, attributes.findEntry(attribute).get());
-                    }
-                });
-            }
-        }).getArtifacts().getArtifactFiles().getFiles();
+        return files;
     }
 
-    private void getDependenciesIdentifiers(Set<ComponentIdentifier> dependenciesIdentifiers, Set<? extends DependencyResult> dependencies) {
-        for (DependencyResult dependency : dependencies) {
-            if (dependency instanceof ResolvedDependencyResult) {
-                ResolvedDependencyResult resolvedDependency = (ResolvedDependencyResult) dependency;
-                ResolvedComponentResult selected = resolvedDependency.getSelected();
-                if (dependenciesIdentifiers.add(selected.getId())) {
-                    // Do not traverse if seen already
-                    getDependenciesIdentifiers(dependenciesIdentifiers, selected.getDependencies());
-                }
-            }
+    @Override
+    public CurrentFileCollectionFingerprint fingerprint(FileCollectionFingerprinter fingerprinter) {
+        if (fingerprint == null) {
+            fingerprint = fingerprinter.fingerprint(files);
```

Comment:
We store the result of the fingerprinting in this class, even though it depends on the argument `fingerprinter`.
That doesn't sound right.

I guess it is for avoiding to calculate the fingerprint in both `hasCachedResult` and `invoke`, right? Can we not do this for now, given that we do not have any performance tests which would actually do some snapshotting?

Reply:
Good point. I made the change.